src/js/setup-text-tracks.js

```diff
@@ -0,0 +1,136 @@
+import dashjs from 'dashjs';
+
+/*
+ * Attach text tracks from dash.js to videojs
+ *
+ * @param {videojs} player the videojs player instance
+ * @param {array} tracks the tracks loaded by dash.js to attach to videojs
+ *
+ * @private
+ */
+function attachDashTextTracksToVideojs(player, tracks) {
+  const mediaPlayer = player.dash.mediaPlayer;
+
+  const trackDictionaryKeyedByTextTrack = new WeakMap();
+
+  // Add remote tracks
+  tracks
+    // Map input data to match HTMLTrackElement spec
+    // https://developer.mozilla.org/en-US/docs/Web/API/HTMLTrackElement
+    .map((track) => Object.assign(
+      {},
+      track,
+      {
+        default: track.defaultTrack,
+        kind: track.kind,
+        label: track.lang,
+        srclang: track.lang,
+      }
+    ))
+
+    // Add track to videojs track list
+    .map((track) => {
+      const remoteTextTrack = player.addRemoteTextTrack(track, true);
+      trackDictionaryKeyedByTextTrack.set(remoteTextTrack.track, track);
```

Comment:
I'm not sure I understand. Later on I use index matching because I have to loop through `player.textTracks()` to find the active track. After that, I use that track to figure out the dash track. `dash.setTextTrack()` expects an index so I have to first find it, then figure out it's index. I must be missing something?